Composer/packages/client/src/recoilModel/dispatchers/lg.ts

```diff
@@ -155,39 +155,53 @@ export const lgDispatcher = () => {
     }) => {
       const { snapshot } = callbackHelpers;
       const lgFiles = await snapshot.getPromise(lgFilesState);
-      let content = lgFiles.find((file) => file.id === id)?.content ?? '';
-      content = lgUtil.updateTemplate(id, content, templateName, template, lgFileResolver(lgFiles)).content;
-      await updateLgFileState(callbackHelpers, { id, content });
+      const lgFile = lgFiles.find((file) => file.id === id);
+      if (!lgFile) {
+        throw new Error(formatMessage('lg file {id} does not exist.', { id }));
+      }
+      const updatedFile = lgUtil.updateTemplate(lgFile, templateName, template, lgFileResolver(lgFiles));
+      await updateLgFileState(callbackHelpers, { id, content: updatedFile.content, updatedFile });
```

Comment:
I think we can use undatedFile to replace the old one. The updateLgFileState will call worker to parse the file content again. So the async can be removed and we can use set function directly.